xarray/core/dataarray.py

```diff
@@ -6706,10 +6707,7 @@ def interp_calendar(
     @_deprecate_positional_args("v2024.07.0")
     def groupby(
         self,
-        group: (
-            Hashable | DataArray | IndexVariable | Mapping[Any, Grouper] | None
-        ) = None,
-        *,
+        group: GroupInput = None,
```

Comment:
```suggestion
        group: GroupInput = None,
        *,
```

Reply:
We've just started the deprecation cycle for this so probably to early

Reply:
It's only the old one, no?

Reply:
Can't tell what you mean.

Reply:
The existing code has the `*`, your proposed changes remove it! 😀

Reply:
Haha sorry! 🤣 I'll address after the weekend if you don't get to it first.